src/Rings/msolve/interface.jl

```diff
@@ -2,6 +2,77 @@
 # an int array lengths storing the lengths of each generator
 # an int array cfs storing the coefficients of each generator
 # an int array exps storing the exponent vectors of each generator
+function convert_oscar_ideal_to_array(
+        I::MPolyIdeal)
+
+    R = base_ring(I)
+    nr_vars     = nvars(R)
+    nr_gens     = ngens(I)
+    field_char  = characteristic(R)
+    lens        = Int32[length(I[i]) for i in 1:nr_gens]
+    nterms      = sum(lens)
+
+    # get coefficients
+    if field_char == 0
+        cfs = BigInt[]
+    else
+        cfs = Int32[]
+    end
+    if field_char == 0
+        for i in 1:nr_gens
+            for cf in coefficients(I[i])
+                push!(cfs, BigInt(numerator(cf)))
+                push!(cfs, BigInt(denominator(cf)))
+            end
+        end
+    else
+        for i in 1:nr_gens
+            for cf in coefficients(I[i])
+                push!(cfs, Int32(cf.data))
```

Comment:
Just so I understand: this makes some specific assumptions about the coefficients, namely that they are from a prime field with small modulus, right? Is this checked or enforced anywhere?

Reply:
Good remark, I've added a check for the size of the prime characteristic.